src/util.ts

```diff
@@ -61,3 +64,39 @@ export function* zip<X, Y>(xs: Iterable<X>, ys: Iterable<Y>): Iterable<X | Y> {
     y1 = y.next()
   }
 }
+
+export async function persistentChatAction(
```

Comment:
Keep only one version of this implementation. The Context method is clearly the superior; I only suggested it as a utility to avoid adding methods to Context. But since this is useful enough, let's keep that.